src/components/ActionBar/src/ActionBarButton.vue

```diff
@@ -49,27 +49,28 @@
 </template>
 
 <script>
-import chroma from 'chroma-js';
+import { colord } from 'colord';
 import PseudoWindow from 'vue-pseudo-window';
 import { MLoading } from '@square/maker/components/Loading';
 import { MThemeKey, defaultTheme, resolveThemeableProps } from '@square/maker/components/Theme';
 import { getContrast } from '@square/maker/utils/get-contrast';
 
-// TODO: refactor the code below so it's shared with Button component
+function setColorVariables(tokens) {
```

Comment:
Would be great to have this as a util maybe `buttonColorVariables` on maker-colors.js so share between this, the Action Bar Button and potentially Choice and Segmented Control? Maybe taking `stateAdjust` as a prop?

Reply:
I thought about that -- but the logic is slightly different right now with the Button since it supports transparent states. I wasn't sure how to support a totally universal utility across all components right now based on the fact that we're conditionally generating different colors based on some internal logic in each component. 

It is definitely possible to generate a more universal utility that could do the same thing for all components. When I started considering a more idealized state perhaps leveraging something like HSLA variables and adjusting output with more CSS, I realized that's a much bigger change than what this task originally set out to do. 

It'd likely be better to tackle that utility separately and look at this PR as a step working us towards that.